lib/elixir/lib/kernel.ex

```diff
@@ -5182,27 +5184,43 @@ defmodule Kernel do
   defmacro sigil_W(term, modifiers)
 
   defmacro sigil_W({:<<>>, _meta, [string]}, modifiers) when is_binary(string) do
-    split_words(string, modifiers)
+    stacktrace = Macro.Env.stacktrace(__CALLER__)
+    split_words(string, modifiers, stacktrace)
   end
 
-  defp split_words(string, []) do
-    split_words(string, [?s])
+  defp split_words(string, [], stacktrace) do
+    split_words(string, [?s], stacktrace)
   end
 
-  defp split_words(string, [mod])
+  defp split_words(string, [mod], stacktrace)
        when mod == ?s or mod == ?a or mod == ?c do
     case is_binary(string) do
       true ->
         parts = String.split(string)
 
+        :lists.foreach(
+          fn part ->
+            if :binary.last(part) == ?, do
+              IO.warn(
+                "item \"#{part}\" in word list has a trailing comma; was this intentional?",
```

Comment:
Suggestion:

> The sigil w does not allow trailing commas at the end of each word. \
> If this was an accident, please remove the comma, otherwise use a regular list.

Reply:
This sounds more like a soft error than a warning, but you're in a position to make that call, so I'm good with it. I'll update the PR when I am near a keyboard.

Reply:
Given that a warning:

1. cannot removed
2. it fails the build on --warning-as-errors

I would say all warnings are "soft errors". It is not the error message that is going to change it, as we don't expect developers to be compiling code with warnings.